core/include/jiminy/core/utilities/random.h

```diff
@@ -536,6 +536,10 @@ namespace jiminy
                                           uint32_t sparsity,
                                           double orientation,
                                           uint32_t seed);
+
+    HeightmapFunction JIMINY_DLLAPI univariatePerlinGround(AbstractPerlinProcess & perlinProcess,
```

Comment:
I'm in favour of instantiate the Perlin process directly inside the method, and pass the construct arguments to this method. It will reduce the set of processes being supported, but I'm not a huge fun of holding a reference to an object that may be modified later on by the user. Plus setting the seed of the Perlin process of an object that is not managed by the method itself is error-prone.

Reply:
So you would prefer something like this?

```
    HeightmapFunction unidirectionalPerlinGround(double wavelength,
                                                std::size_t numOctaves,
                                                 double orientation,
                                                 uint32_t seed)
    {

        const Eigen::Rotation2D<double> rot_mat(orientation);

       

        // Constant for numerical gradient
        const double h = 1e-6;
        const double offset = 100.0;

        return [wavelength, numOctaves, rot_mat, h, offset](
                   const Eigen::Vector2d & pos, double & height, Eigen::Vector3d & normal) -> void
        {
             // Set seed of Perlin Process
            PCG32 pcg32_generator=PCG32(seed);
            perlinProcess = RandomPerlinProcess(wavelength, numOctaves);
            perlinProcess.reset(pcg32_generator);

            // Compute the Perlin Process relative coordinate
            Eigen::Vector2d posRel = (rot_mat.inverse() * pos).array() + offset;

            // Compute the height
            height = perlinProcess(std::abs(static_cast<float>(posRel[0])));

            // Compute the numerical gradient of the Perlin Process
            // and retrieve the normal
            const double curvGrad = 0.5 * (
                perlinProcess(std::abs(static_cast<float>(posRel[0] + h)))
                - perlinProcess(std::abs(static_cast<float>(posRel[0] - h)))
                ) / h;
            normal << -curvGrad, 0.0, 1.0;
            normal.segment<2>(0) = rot_mat * normal.segment<2>(0);
            normal.normalize();
        };
    }
```
(If your forget about the gradient for now of course)
This would also lead to two functions, one for the `RandomPerlinProcess` and one for the `PeriodicPerlinProcess`.

Reply:
Yes. Just make the lambda function an actual function (only visible in `cpp` of course) and use `std::bind` to return a function storing its argument (same as `partial` in python). You could also return a `lambda` calling your method and nothing else, only to capture the arguments.